ibc/host/tests/path_test.go

```diff
@@ -0,0 +1,90 @@
+package tests
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/pokt-network/pocket/ibc/host"
+	coreTypes "github.com/pokt-network/pocket/shared/core/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPaths_GenerateValidIdentifiers(t *testing.T) {
+	ids := make(map[string]string, 100)
+	for i := 0; i < 100; i++ {
+		switch i % 4 {
+		case 0:
+			cl := host.GenerateClientIdentifier()
+			require.NotNil(t, cl)
+			_, ok := ids[cl]
+			require.False(t, ok)
```

Comment:
You use `r := rand.New(rand.NewSource(time.Now().UnixNano()))` as the source of randomness.

This test could potentially get flaky on a fast machine, so I suggest using `i` and set the seed on every iteration. It'll make the test 100% deterministic and 100% non flaky.